hub/package-manager/winget/pinning.md

```diff
@@ -0,0 +1,234 @@
+---
+title: The winget pin command
+description: Use the winget pin command and subcommands to list and manage the repositories Windows Package Manager accesses.
+ms.date: 06/16/2023
+ms.topic: reference
+ms.localizationpriority: medium
+ms.custom: kr2b-contr-experiment
+---
+
+# The winget pin command
+
+The [winget](index.md) tool **pin** command allows you to limit the Windows Package Manager from upgrading a package to specific ranges of versions, or it can prevent it from upgrading a package altogether. A pinned package may still upgrade on its own and be upgraded from outside the Windows Package Manager. 
+
+WinGet pinning supports three types of Package Pinning:
+- **Pinning**:
+The package is excluded from winget upgrade --all but allows winget upgrade <WinGet package>. You can use the "–include-pinned” argument to let winget upgrade --all include pinned packages.
+
+- **Blocking**:
+The package is blocked from winget upgrade --all or winget upgrade <WinGet package>, you will have to unblock the package to let WinGet perform an upgrade.
+
+- **Gating**:
+The package is pinned to specific version(s). For example, if a package is pinned to version 1.2.*, any version between 1.2.0 to 1.2.x is considered valid. To allow user override, “–force” can be used with winget upgrade <WinGet package> to override some of the pinning created above.
```

Comment:
We should add more details about the format for the version pattern. It is way more limited than other similar version range patterns, and people will probably try to use them on winget.